### PR TITLE
Switch VPN to TCP

### PR DESCRIPTION
**Pull request by suhanime**
Currently OpenVPN is set up with UDP, which requires specialized load balancing
This commit configires VPN to use TCP for cloud agnostinc Load Balancing

### CODE REVIEW

File: pkg/aws/helper.go [hunk 1]
```diff
@@ -417,8 +369,8 @@ func (h *AWSHelper) EnsureListener(lbARN, tgARN string, port int, udp bool) erro
 		}
 	}
 	protocol := elbv2.ProtocolEnumTcp
-	if udp {
-		protocol = "UDP"
+	if tcp {
```
Comment (joelddiaz) [inline pkg/aws/helper.go:372]: i think we can keep this `if udp` test b/c it already defaults to TCP already in the line right above, so as long as we call EnsureListener with param `udp == false` this function will continue to work (need to update the EnsureListener call for the VPN in `pkg/aws/install.go` with `udp` set to `false` since we're now using TCP.

or we can remove the `udp` parameter from the function entirely (always assuming TCP), and remove this `if udp` test, and update all the EnsureListener callers.

File: pkg/aws/helper.go [hunk 2]
```diff
@@ -347,55 +347,7 @@ func (h *AWSHelper) RemoveTargetGroup(tgName string) error {
...
-		UnhealthyThresholdCount:    aws.Int64(2),
-	})
-	if err != nil {
-		return "", err
-	}
-	return aws.StringValue(tgResult.TargetGroups[0].TargetGroupArn), nil
-}
-
-func (h *AWSHelper) EnsureListener(lbARN, tgARN string, port int, udp bool) error {
+func (h *AWSHelper) EnsureListener(lbARN, tgARN string, port int, tcp bool) error {
```
Comment (csrwng) [inline pkg/aws/helper.go:350]: The udp flag is no longer needed since now all listeners are only TCP (which is the default). However, I wouldn't make any further changes to this since we're going to get rid of this code once we switch to a service of type LoadBalancer as @joelddiaz is doing here: https://github.com/openshift-hive/hypershift-installer/pull/8

File: pkg/aws/install.go [hunk 3]
```diff
@@ -312,13 +312,13 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
...
 
-	vpnTGARN, err := aws.EnsureUDPTargetGroup(lbInfo.VPC, vpnLBName, vpnNodePort, apiNodePort)
+	vpnTGARN, err := aws.EnsureTargetGroup(lbInfo.VPC, vpnLBName, vpnNodePort)
 	if err != nil {
 		return fmt.Errorf("cannot create VPN target group: %v", err)
 	}
 	log.Infof("Created VPN target group ARN: %s", vpnTGARN)
 
-	if err = aws.EnsureTarget(vpnTGARN, machineID); err != nil {
+	if err = aws.EnsureTarget(vpnTGARN, machineIP); err != nil {
```
Comment (joelddiaz) [inline pkg/aws/install.go:321]: can you explain why the change from machineID to machineIP?

**Review by csrwng — APPROVED** [review]
This looks awesome, thank you!

File: pkg/aws/helper.go (hunk 1 above)
Comment (csrwng) [reply pkg/aws/helper.go:372]: We're about to remove the need for this code, so I don't think we should bother changing this.

File: pkg/aws/install.go (hunk 3 above)
Comment (csrwng) [reply pkg/aws/install.go:321]: AWS UDP load balancers require instance targets instead of IP targets. https://aws.amazon.com/blogs/aws/new-udp-load-balancing-for-network-load-balancer/
> Supported Targets – UDP on Network Load Balancers is supported for Instance target types (IP target types and PrivateLink are not currently supported).

IP targets have less security requirements because they don't require that instances are exposed to the outside world through the security group.
